mountpoint-s3/src/cli.rs

```diff
@@ -286,20 +287,39 @@ pub struct CliArgs {
         value_parser = clap::builder::NonEmptyStringValueParser::new(),
     )]
     pub sse_kms_key_id: Option<String>,
+
+    #[clap(
+        long,
+        help = "Disable S3 additional checksums for object uploads",
+        help_heading = BUCKET_OPTIONS_HEADER,
+    )]
+    pub disable_upload_checksums: bool,
```

Comment:
I'm a little uncertain about this one - do we want a boolean, or would we rather have some kind of mode parameter like `--trailing-checksums <none|crc32c>`? Today we use CRC-32C or turn them off, but we could allow customers to choose between additional checksums (SHA-1, SHA-256, CRC-32) in the future if we really wanted to.

Maybe it provides a path for Outpost customers to opt-in to checksums without being forced to upgrade to get checksums too if they became available, as discussed here: https://github.com/awslabs/mountpoint-s3/pull/849/files#r1565606507?

We could have `--disable-upload-checksums` easily alias to `--trailing-checksums none` in the future if we did make that change, so I don't have a strong preference here.

Reply:
Yeah, I like this idea, done.